### PR TITLE
Avoid unnecessary merklisation of initial deposits

### PR DESCRIPTION
**Pull request by mkalinin**

### CODE REVIEW

File: consensus/src/main/java/org/ethereum/beacon/consensus/spec/GenesisFunction.java [hunk 1]
```diff
@@ -55,18 +55,14 @@ default BeaconState initialize_beacon_state_from_eth1(
...
-      process_deposit(state, deposit);
-    }
+    ReadList<Integer, DepositData> deposit_data_list =
+        ReadList.wrap(
+            deposits.stream().map(Deposit::getData).collect(Collectors.toList()),
+            Integer::new,
+            1L << getConstants().getDepositContractTreeDepth().getIntValue());
+    state.setEth1Data(new Eth1Data(
+        hash_tree_root(deposit_data_list), UInt64.valueOf(deposits.size()), eth1_block_hash));
+    deposits.forEach(deposit -> process_deposit(state, deposit));
```
Comment (zilm13) [inline consensus/src/main/java/org/ethereum/beacon/consensus/spec/GenesisFunction.java:70]: I'd add comment why verification is omitted

**Review by zilm13 — APPROVED** [review]
LGTM

**Review by ericsson49 — APPROVED** [review]
LGTM.

File: consensus/src/main/java/org/ethereum/beacon/consensus/spec/GenesisFunction.java (hunk 1 above)
Comment (mkalinin) [reply consensus/src/main/java/org/ethereum/beacon/consensus/spec/GenesisFunction.java:70]: Done!

**Pull request merged**